schema/compose-spec.json

```diff
@@ -606,15 +611,26 @@
               "items": {
                 "type": "object",
                 "properties": {
-                  "subnet": {"type": "string"}
-                },
-                "additionalProperties": false,
-                "patternProperties": {"^x-": {}}
-              }
+                  "subnet": {"type": "string", "format": "subnet_ip_address"},
+                  "ip_range": {"type": "string"},
+                  "gateway": {"type": "string"},
+                  "aux_addresses": {
+                    "type": "object",
+                    "additionalProperties": false,
+                    "patternProperties": {"^.+$": {"type": "string"}}
+                  }
+                }
+              },
+              "additionalProperties": false,
+              "patternProperties": {"^x-": {}}
+            },
+            "options": {
+              "type": "object",
+              "additionalProperties": false,
+              "patternProperties": {"^.+$": {"type": "string"}}
             }
           },
-          "additionalProperties": false,
-          "patternProperties": {"^x-": {}}
+          "additionalProperties": false
```

Comment:
Shouldn't we keep `patternProperties` here or is the removal intentional? This means that no custom/experimental fields can be attached to `ipam` config.

Reply:
My mistake, I'll add it. Thanks for noticing it!

Reply:
nice catch!